File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = ''
    stack = []

    for num in number:
        while stack and stack[-1] < num and k>0:    # 제거 조건 충족
            stack.pop(-1)
            k -=1
        stack.append(num)
    
    if k > 0:
        stack = stack[:-k]
            
    answer = ''.join(stack)
    return answer

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    const stack = [number[0]];
    for(let i=1; i<number.length; i++){
        while (true){
            if(stack.length > 0 && k>0 && stack[stack.length-1]<number[i]) {
                stack.pop();
                k--;
            } else{
                stack.push(number[i]);
                break;
            }
        }
    }

    return stack.slice(0,stack.length-k).join("");
}